core/init/db/db.go

```diff
@@ -1,89 +1,13 @@
 package db
 
 import (
-	"fmt"
-	"log"
-	"os"
 	"path"
-	"time"
 
 	"github.com/1Panel-dev/1Panel/core/global"
-	"github.com/glebarez/sqlite"
-	"gorm.io/gorm"
-	"gorm.io/gorm/logger"
+	"github.com/1Panel-dev/1Panel/core/utils/common"
 )
 
 func Init() {
-	initDB()
-	initTaskDB()
-}
-
-func initDB() {
-	dbPath := path.Join(global.CONF.System.BaseDir, "1panel/db")
-	if _, err := os.Stat(dbPath); err != nil {
-		if err := os.MkdirAll(dbPath, os.ModePerm); err != nil {
-			panic(fmt.Errorf("init db dir failed, err: %v", err))
-		}
-	}
-	fullPath := path.Join(dbPath, global.CONF.System.DbCoreFile)
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.DB = db
-	global.LOG.Info("init db successfully")
-}
-
-func initTaskDB() {
-	fullPath := path.Join(global.CONF.System.BaseDir, "1panel/db/task.db")
-	if _, err := os.Stat(fullPath); err != nil {
-		f, err := os.Create(fullPath)
-		if err != nil {
-			panic(fmt.Errorf("init task db file failed, err: %v", err))
-		}
-		_ = f.Close()
-	}
-
-	db, err := NewDBWithPath(fullPath)
-	if err != nil {
-		panic(err)
-	}
-
-	global.TaskDB = db
-	global.LOG.Info("init task db successfully")
-}
-
-func NewDBWithPath(dbPath string) (*gorm.DB, error) {
-	db, _ := gorm.Open(sqlite.Open(dbPath), &gorm.Config{
-		DisableForeignKeyConstraintWhenMigrating: true,
-		Logger:                                   getLogger(),
-	})
-	sqlDB, dbError := db.DB()
-	if dbError != nil {
-		return nil, dbError
-	}
-	sqlDB.SetConnMaxIdleTime(10)
-	sqlDB.SetMaxOpenConns(100)
-	sqlDB.SetConnMaxLifetime(time.Hour)
-	return db, nil
-}
-func getLogger() logger.Interface {
-	return logger.New(
-		log.New(os.Stdout, "\r\n", log.LstdFlags),
-		logger.Config{
-			SlowThreshold:             time.Second,
-			LogLevel:                  logger.Silent,
-			IgnoreRecordNotFoundError: true,
-			Colorful:                  false,
-		},
-	)
+	global.DB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/core.db"), "core")
+	global.TaskDB = common.LoadDBConnByPath(path.Join(global.CONF.System.BaseDir, "1panel/db/task.db"), "task")
 }
```

Comment:
To summarize, there is no significant difference between the provided PHP code snippet and the Go code snippets you sent (Init function definitions). However, I can make several notes based on those:

**Note**: The comments suggest that SQLite databases (`sqlite`). To simplify this implementation, GORM uses PostgreSQL by default but it seems like the database connection strings were not properly set to use sqlite.

The main differences are:
1. In Go code, an `os` module was used instead of a `fileutil`. Also, some functions from both languages appear redundant given their similarities.
2. There’s missing docstrings which could be helpful for future reference, especially for developers switching across programming languages.
3. The initializers look incomplete with the lack of required parameters in each function definition.

For optimization purposes, consider adding appropriate docstrings to explain how these scripts should operate. They seem mostly operational and work without any major issues when run directly. But it's recommended to verify all environment-specific details such as permissions or settings within specific project contexts.

In terms of code complexity, it appears that most of its operations are straightforward and maintainable; however, using external libraries like GORM might require more familiarity compared to local MySQL implementations.

Lastly, ensure that proper logging mechanisms are implemented, since debugging errors related to DB connections requires detailed logs. Without further configuration, it may still show warnings indicating the inability to connect to the database due to permissions.